src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -357,7 +411,7 @@ struct sigaction* get_chained_signal_action(int sig) {
   }
   if (actp == NULL) {
     // Retrieve the preinstalled signal handler from jvm
-    actp = get_preinstalled_handler(sig);
+    actp = const_cast<struct sigaction*>(chained_handlers.get(sig));
```

Comment:
Must `SavedSignalHandlers::get()` have the **const struct** in `const struct sigaction* get(int sig) const` signature?

If it was just `struct sigaction* get(int sig) const` then we wouldn't need this awkward cast.

Reply:
I'm a stickler to const since its very expressive and prevents stupid errors. `const struct sigaction* get() const` makes perfect sense since at no time you are supposed to modify the underlying sigaction structure. Once it is saved, it should only be used to read from (eg to print, or to extract the handler address and call it in `call_chained_handler()`.

But old code was not const correct. The way to start being const correct without having to rewrite everything is to make new code const correct and cauterize at the interface between new and old code with these ugly but at least expressive casts.

Side note, even the sigaction function is const correct: https://pubs.opengroup.org/onlinepubs/007904875/functions/sigaction.html
see the `act` input parameter designating the new handler, its const. Only the output parameter `oact` is nonconst.

I will see if I can make the rest of the code const correct too wrt the sigaction structures. That way we can get rid of the cast and have better code as well.

Reply:
If we can do that, then that would be ideal.